### PR TITLE
Pr ts feature/cascading select

### LINKED ISSUE #206: Add support for JIRA cascading custom field
### Describe the problem

Currently CxFlow does not support JIRA cascading custom fields.

### Proposed solution

Add a new custom field type for the JIRA service that is associated with the JIRA cascading custom field.

### PR DESCRIPTION
**Pull request by kmcdon83**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](https://github.com/checkmarx-ts/open-source-template/blob/master/CODE-OF-CONDUCT.md). Please see the [contributing guidelines](https://github.com/checkmarx-ltd/open-source-template/blob/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

This change is to support a new custom field of cascading select within JIRA.  Implementation was completed for a customer and changes are being pulled into main code base.

### References

Fixes #206 

### Testing

Changes are minimal and non-impactful.  Changes were tested with manual validation leveraging custom field of this type.

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentaiton is a Wiki Update, please indicate desired changes within PR MD Comment*
_Wiki creation is in process_
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 1]
```diff
@@ -627,6 +628,22 @@ private void mapCustomFields(ScanRequest request, ScanResults.XIssue issue, Issu
...
+                            // must match case
+                            String[] selectedValues = StringUtils.split(value, ";");
+                            if(selectedValues.length == 2) {
+                                Map<String, Object> cascadingValues = new HashMap<String, Object>();
+                                cascadingValues.put(VALUE_FIELD_TYPE, selectedValues[0].trim());
+                                cascadingValues.put(CHILD_FIELD_TYPE, ComplexIssueInputFieldValue.with(VALUE_FIELD_TYPE, selectedValues[1].trim()));
+                                issueBuilder.setFieldValue(customField, new ComplexIssueInputFieldValue(cascadingValues));
+                            }
+                            else {
+                                log.warn("Invalid value for jira field type {}", f.getJiraFieldType());
```
[the remark this answers is not in the document]
Comment (kmcdon83) [reply src/main/java/com/checkmarx/flow/service/JiraService.java]: Done

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 2]
```diff
@@ -627,6 +628,22 @@ private void mapCustomFields(ScanRequest request, ScanResults.XIssue issue, Issu
                             }
                             issueBuilder.setFieldValue(customField, fields);
                             break;
+                        case "cascading-select":
```
[the remark this answers is not in the document]
Comment (kmcdon83) [reply src/main/java/com/checkmarx/flow/service/JiraService.java:632]: Done

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 3]
```diff
@@ -627,6 +628,22 @@ private void mapCustomFields(ScanRequest request, ScanResults.XIssue issue, Issu
...
+                            log.debug("cascading select list field");
+                            // expected value format is "parent;child"
+                            // neither can be empty; enclose in quotes if spaces/special characters
+                            // must match case
+                            String[] selectedValues = StringUtils.split(value, ";");
```
[the remark this answers is not in the document]
Comment (kmcdon83) [reply src/main/java/com/checkmarx/flow/service/JiraService.java]: Done

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1000127&projectid=31)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|554
Low|5378
Informational|30

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

**Comment by kmcdon83** [comment]
@natalieb07 Seems like the quality gate is failing on old code here.

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
### Checkmarx scan completed
[Full Scan Details](http://EC2AMAZ-JP2I9QF/CxWebClient/ViewerMain.aspx?scanid=1000131&projectid=31)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|549
Low|5866
Informational|44

#### Violation Summary
Severity|Count
---|---

#### Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---

**Comment by kmcdon83** [comment]
This PR got ugly.  I am going to close it and introduce changes cleanly.  I didn't realize it never made it in.

**Pull request closed without merging**